doc/policy/packages.md

```diff
@@ -57,3 +57,40 @@ test accepts):
 
    - Warning: Batched fee-bumping may be unsafe for some use cases. Users and application developers
      should take caution if utilizing multi-parent packages.
+
+### Package Fees and Feerate
+
+*Package Feerate* is the total modified fees divided by the total virtual size of all transactions
+in the package after deduplication.
+
+To meet the two feerate requirements of a mempool, i.e., the pre-configured minimum relay feerate
+(`minRelayTxFee`) and dynamic mempool minimum feerate, the total package feerate is used instead of
+the individual feerate. The individual transactions are allowed to be below feerate requirements if
+the package meets the feerate requirements. For example, the parent(s) in the package can have 0
+fees but be paid for by the child.
+
+*Rationale*: This can be thought of as "CPFP within a package," solving the issue of a parent not
+meeting minimum fees on its own. This allows L2 applications to adjust their fees at broadcast time
+instead of overshooting or risking getting stuck/pinned.
+
+*Rationale*: It would be incorrect to use the fees of transactions that are already in the mempool, as
+we do not want a transaction's fees to be double-counted.
+
+### Transactions Are Always Submitted Individually First
+
+Transactions within a package are always validated individually first, and package validation is
+used for the transactions that failed.
+
+*Rationale*: Packages are intended for incentive-compatible fee-bumping. That is, transaction A is a
+"legitimate" fee-bump transaction B only if A is a descendant of B and has a higher feerate.
+Package feerate, as defined above, does not consider the fees and relationships between all
+transactions in the package. We want to prevent "parents pay for children" behavior; fees of
+children should have no impact on the consideration of their parents, since they are not necessary
+for the parents to be mined. More generally, if transaction B is not needed in order for transaction
+A to be mined, B's fees cannot help A. In a child-with-parents package, simply excluding any parent
+transactions that meet feerate requirements individually is sufficient to ensure this.
+
+*Rationale*: For backwards-compatibility of transaction relay policy, it's important that package
```

Comment:
i agree as a design principle that backwards-compatibility of transaction relay policy is worthwhile.

That said, we might have in the future to make exceptions to that principle, where a transaction/package might be relayed across non-upgraded nodes though rejected by upgraded nodes. E.g I'm thinking if replace-by-feerate is deployed instead of replace-by-fee, you might have a (2000 sats; 10 sat/vB) transaction replacing a (5000 sats; 5 sat/vB) already-in-mempool one (even if we can envision some transition period with dual-support).

Thus I'm suggesting to lightweight a bit the wording here, "As a principle, backwards-compatibility of transaction relay policy is prioritized", "we prevent non-programmed restriction of policy". Especially as all those questions of policy loosening/tightening and how the project operates them are relatively young, I think it's cautious to be more formal in the phrasing. Thereby avoiding in the future ecosystems users to claim an adamant stability of Core policy rules, like in the zero-conf/full-rbf past polemics.

Reply:
Agree, added a sentence at the beginning to this effect.